tools/fragpipe/fragpipe.xml

```diff
@@ -0,0 +1,276 @@
+<tool id="fragpipe" name="FragPipe -  Academic Research and Education User License (Non-Commercial)" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="20.05">
+    <description>Data analysis for mass spectrometry-based proteomics</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+
+    <stdio>
+        <!-- The error code from individual tools run by FragPipe is not always propogated to the FragPipe process exit code -->
+        <regex match="exit code: [^0]"
+               source="stdout"
+               level="fatal"
+               description="A FragPipe pipeline process returned non-zero exit code."/>
+    </stdio>
+
+    <version_command>fragpipe --help | grep 'FragPipe v'</version_command>
+
+    <!--
+        FragPipe wrapper command:
+        - Links input files
+        - Modifies database path in workflow file
+        - Runs Philosopher to add decoys and contaminants
+        - Runs FragPipe, passing paths to MSFragger, Philosopher, Python, and IonQuant
+    -->
+    <command detect_errors="exit_code"><![CDATA[
+        ## Create symlink for workflow output
+        ln -s $workflow_configfile fp.workflow;
+
+        ## The user may opt to only output the workflow to review the parameters. If they haven't selected this option, we prepare and run FragPipe.
+        #if not $workflow_only
+            @PREPARE_ENVIRONMENT@
+            @PREPARE_INPUTS@
+            @PREPARE_DATABASE@
+
+            ## Run FragPipe
+            '$__tool_directory__/fragpipe'
+                --msfragger_key f28c4be2-759b0a11-6c1ea652-1240565a
+                --ionquant_key 47cb2953-b4bf7726-2c96f5fe-8f1013aa
+                --headless
+                --threads \${GALAXY_SLOTS:-8}
+                --ram \$[ \${GALAXY_MEMORY_MB:-8192} / 1024 ]
+                --workflow fp.workflow --manifest fp.manifest
+                --workdir outputs
+                --config-msfragger \${CONDA_PREFIX}/share/msfragger-*/MSFragger-*/MSFragger-*.jar
+                --config-philosopher `which philosopher`
+                --config-python `which python`
+                --config-ionquant \${CONDA_PREFIX}/share/ionquant-*/IonQuant.jar &&
```

Comment:
I'm surprised that CONDA_PREFIX works in a container setup. Does `which philosopher/../share/ionquant` also works and might be more stable?

Reply:
ping ... The conda prefix is what makes me nervous in container envs

Reply:
Sorry for the delay - I just pushed a commit that removes all CONDA_PREFIX references.